src/modules/todo/components/todo/Todo.tsx

```diff
@@ -1,34 +1,33 @@
-import type { TodoItem } from './Todo.interface'
 import { useForm } from 'react-hook-form'
-import { useFetchTodos } from '../../../../hooks/useFetchTodos'
 import * as classes from './Todo.module.scss'
+import { useMutationToPostTodo } from '@/modules/todo/hooks/useMutationToPostTodo'
+import { useFetchTodos } from '@/modules/todo/hooks/useFetchTodos'
 
 type AddTodoFormValues = {
   text: string
 }
 
 function Todo() {
-  const [todos, setTodos] = useFetchTodos()
-
   const { register, handleSubmit, reset } = useForm<AddTodoFormValues>()
 
-  const onSubmit = handleSubmit((data) => {
-    fetch('/api/todos', { method: 'POST', body: JSON.stringify(data) })
-      .then(async (res) => {
-        const { todo } = (await res.json()) as { todo: TodoItem }
-        setTodos((prev) => [...prev, todo])
-        reset()
-      })
-      .catch((err) => {
-        console.error(err)
-      })
+  const todoList = useFetchTodos()
+
+  const mutation = useMutationToPostTodo()
```

Comment:
Same with namin, mutation isn't clear what exatly we should be doing here. it's better to give some meaningfull name like `todoCreate`, you can do it two ways, first one:
`const todoCreate = useTodoCreate()`
second one:
`const {mutate: todoCreate} = useTodoCreate()`
also same with naming for the hook itself it's better to use `useTodoCreate`